presto-orc/src/test/java/com/facebook/presto/orc/TestReadBloomFilter.java

```diff
@@ -0,0 +1,133 @@
+import static com.facebook.presto.orc.metadata.CompressionKind.SNAPPY;
+import static com.facebook.presto.spi.predicate.TupleDomain.fromFixedValues;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.SmallintType.SMALLINT;
+import static com.facebook.presto.spi.type.TinyintType.TINYINT;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.collect.Iterables.cycle;
+import static com.google.common.collect.Iterables.limit;
+import static com.google.common.collect.Lists.newArrayList;
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+
+public class TestReadBloomFilter
+{
+    private static final TimeZoneKey TIME_ZONE = TimeZoneKey.getTimeZoneKey(HIVE_STORAGE_TIME_ZONE.getID());
+
+    @Test
+    public void test()
+            throws Exception
+    {
+        testType(TINYINT, ImmutableList.of(1L, 50L, 100L), 50L, 77L);
+        testType(SMALLINT, ImmutableList.of(1L, 5000L, 10_000L), 5000L, 7777L);
+        testType(INTEGER, ImmutableList.of(1L, 500_000L, 1_000_000L), 500_000L, 777_777L);
+        testType(BIGINT, ImmutableList.of(1L, 500_000L, 1_000_000L), 500_000L, 777_777L);
+
+        // Bloom Filters are not supported for DECIMAL, FLOAT, DATE, TIMESTAMP, and CHAR
+
+        testType(DOUBLE, ImmutableList.of(1.11, 500_000.55, 1_000_000.99), 500_000.55, 777_777.77);
+
+        testType(VARCHAR, ImmutableList.of("a", "o", "z"), utf8Slice("o"), utf8Slice("w"));
+        testType(VARBINARY,
+                ImmutableList.of(new SqlVarbinary("a".getBytes(UTF_8)), new SqlVarbinary("o".getBytes(UTF_8)), new SqlVarbinary("z".getBytes(UTF_8))),
+                utf8Slice("o"),
+                utf8Slice("w"));
+    }
+
+    private static <T> void testType(Type type, List<T> uniqueValues, T inBloomFilter, T notInBloomFilter)
+            throws Exception
+    {
+        ArrayList<T> writeValues = newArrayList(limit(cycle(uniqueValues), 30_000));
+
+        try (TempFile tempFile = new TempFile()) {
+            writeOrcColumnHive(tempFile.getFile(), ORC_12, SNAPPY, type, writeValues);
+
+            // without predicate a normal block will be created
+            try (OrcBatchRecordReader recordReader = createCustomOrcRecordReader(tempFile, OrcPredicate.TRUE, type, MAX_BATCH_SIZE)) {
+                assertEquals(recordReader.nextBatch(), 1024);
+            }
+
+            // predicate for specific value within the min/max range without bloom filter being enabled
+            TupleDomainOrcPredicate<String> noBloomFilterPredicate = new TupleDomainOrcPredicate<>(
```

Comment:
the logic of building the predicate repeats multiple time; consider extracting a helper method

```
    private static <T> TupleDomainOrcPredicate<String> makeOrcPredicate(Type type, T value, boolean bloomFilterEnabled)
    {
        return new TupleDomainOrcPredicate<>(
                fromFixedValues(ImmutableMap.of("test", NullableValue.of(type, value))),
                ImmutableList.of(new ColumnReference<>("test", 0, type)),
                bloomFilterEnabled,
                Optional.empty());
    }
```